com.unity.multiplayer.mlapi/Runtime/Timing/NetworkTickSystem.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using Unity.Profiling;
+using UnityEngine;
+
+namespace MLAPI.Timing
+{
+    public class NetworkTickSystem
+    {
+#if DEVELOPMENT_BUILD || UNITY_EDITOR
+        private static ProfilerMarker s_Tick = new ProfilerMarker($"{nameof(NetworkTimeSystem)}.Tick");
+#endif
+
+        /// <summary>
+        /// Special value to indicate "No tick information"
+        /// </summary>
+        public const int NoTick = int.MinValue;
+
+        /// <summary>
+        /// The TickRate of the tick system. This is used to decide how often a fixed network tick is run.
+        /// </summary>
+        public int TickRate { get; }
+
+        /// <summary>
+        /// The current local time. This is the time at which predicted or client authoritative objects move. This value is accurate when called in Update or during the <see cref="Tick"/> event but does not work correctly for FixedUpdate.
+        /// </summary>
+        public NetworkTime LocalTime { get; internal set; }
+
+        /// <summary>
+        /// The current server time. This value is mostly used for internal purposes and to interpolate state received from the server. This value is accurate when called in Update or during the <see cref="Tick"/> event but does not work correctly for FixedUpdate.
+        /// </summary>
+        public NetworkTime ServerTime { get; internal set; }
+
+        /// <summary>
+        /// Gets invoked before every network tick.
+        /// </summary>
+        public event Action Tick;
```

Comment:
nit
```suggestion
        public event Action OnTick;
```

Reply:
So from my understanding of C# naming conventions the `On` prefix is used for a function which raises an event but not the event itself. Not sure if I'm right about this as I see people using different conventions for this all the time. https://stackoverflow.com/questions/724085/events-naming-convention-and-style

Reply:
Good point, I wonder what Unity itself and other Unity packages are using. We might just simply follow their standards as well here.